include/nautilus/cpuid.h

```diff
@@ -289,13 +289,15 @@ typedef struct cpuid_ret {
 static int
 cpuid (uint32_t func, cpuid_ret_t * ret)
 {
+    #ifdef NAUT_CONFIG_ARCH_X86
```

Comment:
Same as above. We should do this hack at the user site, not the implementation site